Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Action/ActionManagerInterface.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Interface/Interface.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/std/function/function_base.h>
+
+#include <QWidget>
+
+class QAction;
+
+namespace AzToolsFramework
+{
+    using ActionManagerOperationResult = AZ::Outcome<void, AZStd::string>;
+
+    //! ActionManagerInterface
+    //! Interface to register and trigger actions in the Editor.
+    class ActionManagerInterface
```

Comment:
suggestion: As you want to inherit from this, you actually might want to add the virtual destructor here (sorry I missed that before), if you don't want to delete the object from a base pointer you can make the destructor protected and non-virtual (which is the preferred approach for EBuses)

Reply:
I have seen that approach for EBuses, but not for Interfaces - is it actually used somewhere?
I'm likely going to move on with this submission as is, but I can definitely fix this up in a follow-up.